spec/predicates/vsa.md

```diff
@@ -1,11 +1,279 @@
+Version: 1.1
 
 ## Purpose
 
-Records a SLSA verification decision about an artifact.
+Verification summary attestations communicate that an artifact has been verified
+against some policy and details about that verification. Such details may
+include, but are not limited to, what SLSA level the artifact and/or its
+dependencies were verified to meet.
 
-For more details, please see https://slsa.dev/spec/v1.0/verification_summary
+## Use Cases
+
+Software consumers make a decision about the validity of an
+artifact without needing to have access to all of the attestations about the
+artifact or all of its transitive dependencies.  They delegate
+complex policy decisions to some trusted party and then simply trust that
+party's decision regarding the artifact. That decision is represented by a VSA.
+
+VSAs also allow software producers to keep the details of their build pipeline
+confidential while still communicating that some verification has taken place.
+This might be necessary for legal reasons (keeping a software supplier
+confidential) or for security reasons (not revealing that an embargoed patch has
+been included).
+
+## Prerequisites
+
+Understanding of [in-toto specification] and the in-toto attestation framework.
+
+## Model
+
+A Verification Summary Attestation (VSA) is an attestation that some entity
+(`verifier`) verified one or more software artifacts (the `subject` of an
+in-toto attestation [Statement]) by evaluating the artifact and a `bundle`
+of attestations against some `policy`.  Users who trust the `verifier` may
+assume that the artifacts satisfied the indicated security polciy without
+themselves needing to evaluate the artifact or to have access to the
+attestations the `verifier` used to make its determination.
+
+The VSA also allows consumers to determine the verified levels of
+all of an artifact’s _transitive_ dependencies.  The verifier does this by
+either a) verifying the provenance of each non-source dependency listed in
+the [resolvedDependencies](/provenance/v1#resolvedDependencies) of the artifact
+being verified (recursively) or b) matching the non-source dependency
+listed in `resolvedDependencies` (`subject.digest` ==
+`resolvedDependencies.digest` and, ideally, `vsa.resourceUri` ==
+`resolvedDependencies.uri`) to a VSA _for that dependency_ and using
+`vsa.verifiedProperties` and `vsa.dependencyProperties`.  Policy verifiers
+wishing to establish minimum requirements on dependencies may use
+`vsa.dependencyProperties` to do so.
+
+## Schema
+
+```jsonc
+// Standard attestation fields:
+"_type": "https://in-toto.io/Statement/v1",
+"subject": [{
+  "name": <NAME>,
+  "digest": { <digest-in-request> }
+}],
+
+// Predicate
+"predicateType": "https://slsa.dev/verification_summary/v1",
+"predicate": {
+  "verifier": {
+    "id": "<URI>",
+    "version": {
+      "<COMPONENT>": "<VERSION>",
+      ...
+    }
+  },
+  "timeVerified": <TIMESTAMP>,
+  "resourceUri": <artifact-URI-in-request>,
+  "policy": {
```

Comment:
It is already a ResourceDescriptor (but not a list) according to the doc below.  I guess we just need to update this summary?

Reply:
Or `"policy": #ResourceDescriptor,` to match https://slsa.dev/spec/v1.0/provenance#schema

Reply:
@adityasaky , I agree that it makes sense to support lists of policies. Even in the SLSA use case, there may be specific individual policies for each track. It would be beneficial to be able to aggregate all policies into a single VSA. There could additionally be policies that are owned/maintained by various organizations -- being able to differentiate between them (and their results) separately seems like a good direction for transparency.

Reply:
A list would be nice, but the same could also be achieved by creating multiple VSAs. We should explore which option makes the most sense.

Reply:
@kpk47 In the case of Witness policies (and I think in-toto layouts), you would have keys to sign the policy itself, and potential keys for each functionary that performed an operation. The later could be embedded in the policy but, the signing key for the policy itself is an out-of-band communication (needs a separate resource description).

As far as multiple policies go... I think it's going to get complicated fast. What logic is applied between multiple policies? You could defer that decision making process external to the VSA if you're only capturing discrete results per policy but, for me, that's counter to the purpose of capturing a policy decision in a VSA. I think that @lcarva is on the write track with additional VSAs. Each could be inputs to a subsequent policy or future decision.

Reply:
> What logic is applied between multiple policies?

That would depend on the policy engine responsible for creating the VSA. If multiple policies are given as input to a decision, the VSA-generating tool would be responsible for reconciling any conflicts/inconsistencies and generating the ultimate pass/fail. Even if a policy engine supports multiple policies as input, it might elect to create multiple VSAs. 

> You could defer that decision making process external to the VSA if you're only capturing discrete results per policy but, for me, that's counter to the purpose of capturing a policy decision in a VSA.

Depending on the cross-relationship between different policies for the policy engine, this might not be straightforward. Another potential approach could be for an engine to create pseudo policies that are composed of multiple interrelated sub policies and just report _that_. One potential use case for multiple policies would be if there were policy modifications on-the-fly, i.e. using inline policies as mentioned by @lcarva: https://github.com/in-toto/attestation/pull/295#discussion_r1414559935.

Reply:
Apologies for the delay, my notifications have been a mess the last 3 weeks or so.

@jkjell covered my point about policy keys above. I'm actually less convinced a VSA needs to cover multiple actual policies. I think a verification workflow should be scoped per policy artifact (and applicable attestations), so if multiple policies are verified, we should generate multiple VSAs. This, to me, makes sense given the expected use cases for VSAs: record of verification elsewhere in the supply chain or for policy delegations like [in-toto's sublayouts](https://github.com/in-toto/docs/blob/master/in-toto-spec.md#45-specifying-sublayouts).